oscar_stripe_sca/facade.py

```diff
@@ -482,6 +508,146 @@ def capture_order_payment(self, order_number, **kwargs):
+                        "order_number": order_number,
+                    }
+                })
+                coupons.append(coupon)
+
+        # And we finally create the invoice.
+        self.logger.info(
+            f"*** Creating invoice for checkout session: {checkout_session.id}"
+        )
+        today = round(dt.now(tz.utc).timestamp())
+        number = self._get_next_invoice_number()  # overridden in the project
+        customer = checkout_session.customer
+        params = {
+            "number": number,
+            "customer": customer,
+            "collection_method": "send_invoice",
+            "due_date": today,
+            "automatic_tax": {
+                "enabled": settings.STRIPE_ENABLE_TAX_COMPUTATION,
+            },
+        }
+        if coupons:
+            params.update(
+                {"discounts": [
+                    {"coupon": coupon.id} for coupon in coupons
+                ]}
+            )
+        invoice = invoicer.create(params=params)
+
+        invoice_id = invoice.id
+        invoice_number = invoice.number
+        self.logger.debug(f"*** invoice_id: {invoice_id}")
+        self.logger.debug(f"*** invoice_number: {invoice_number}")
+
+        # We now add lines to the invoice, for each one of the order.
+        invoice_info = f"{invoice_id} (number: {invoice.number})"
+        self.logger.info(f"*** Adding lines to invoice: {invoice_info}")
+        invoice_lines = []
+        order_lines = order.lines.all()
+        for order_line in order_lines:
+            product = order_line.product
+            product_type = product.get_display_type()
+            product_title = product.get_display_title()
+            description = f"[{product_type}] {product_title}"
+            amount = int(order_line.unit_price_excl_tax *100)
+            invoice_lines.append({
+                "description": description,
+                "amount": amount,
+            })
+        params={"lines": invoice_lines}
+        invoicer.add_lines(invoice=invoice_id, params=params)
+
+        # The invoice may now be finalized...
+        self.logger.info(f"*** Finalizing invoice: {invoice_info}")
+        invoicer.finalize_invoice(invoice=invoice_id)
+
+        # ... linked to its payment...
+        self.logger.info(f"*** Attaching payment to invoice: {invoice_info}")
+        params = {"payment_intent": payment_intent_id}
+        invoicer.attach_payment(invoice_id, params=params)
+
+        # ... and sent, *if* that should be done through Stripe.
+        if settings.STRIPE_INVOICE_SENDING == INVOICE_SENDING_AUTOMATIC:
+            self.logger.info(f"*** Sending invoice: {invoice_info}")
+            invoicer.send_invoice(invoice_id)
+
+        return invoice_id
+
+    def retrieve_invoice(self, invoice_id):
+        return self.stripe_client.invoices.retrieve(invoice_id)
+
+    def record_invoice(self, invoice_id, payment_intent_id):
+        pass  # Customize at will!
```

Comment:
If this is optional `pass` is OK. Otherwise better to `raise NotImplementedError`